### DISCUSSION TITLE
Add support for navigation destination

### DISCUSSION THREAD

**Discussion by longzheng** [teslamate-org/teslamate#3250] [Ideas]
In the latest version of the Tesla API, they actually return the destination name and metadata if one is set in the navigation.

```json
        "drive_state": {
            "active_route_destination": "Home",
            "active_route_energy_at_arrival": 73,
            "active_route_latitude": -1.123456,
            "active_route_longitude": 1.123456,
            "active_route_miles_to_arrival": 6.485299,
            "active_route_minutes_to_arrival": 23.466667,
            "active_route_traffic_minutes_delay": 0.0,
            "gps_as_of": 1686894025,
            "heading": 254,
            "latitude": -1.123456,
            "longitude": 1.123456,
```

It would be good to log this data as part of a trip, and expose this data via MQTT (so it can be consumed by Home Assistant for automations).

**Comment by johnbasl** [comment]
Yes, this would be excellent. I've been trying to find a way to trigger events when I navigate to certain locations.

**Comment by earthsea** [comment]
I agree. In an integrated HomeAssistant and Teslamate, we can use the vehicle's destination information to accurately trigger the heating and cooling of the home.

**Comment by vincep5** [comment]
I'd also like to have this information. In combination with Home Assistant and MQTT I could automate some home things!

**Comment by longzheng** [comment]
I added support for this in https://github.com/teslamate-org/teslamate/pull/3657

Will look forward to it in the next release.

**Comment by longzheng** [comment]
For anyone else that was interested in this, it's now available in version 1.28.4 https://github.com/teslamate-org/teslamate/releases/tag/v1.28.4

The MQTT topics are
```
teslamate/cars/$car_id/active_route_destination
teslamate/cars/$car_id/active_route_latitude
teslamate/cars/$car_id/active_route_longitude
```

**Reply by brianmay** [reply]
This might change, see #3789

Feedback appreciated.
